app/views/shared/authentication/_forem_creator_signup.html.erb

```diff
@@ -1,3 +1,96 @@
+      <% if flash[:notice] %>
+        <div class="crayons-notice crayons-notice--danger mb-6" role="alert">
+          <%= flash[:notice] %>
+        </div>
+      <% end %>
+
+      <% any_errors = local_assigns[:resource]&.errors&.any? %>
+
+      <%= form_for(@user, as: :user, data: { testid: "creator-signup-form" }, url: registration_path(:user)) do |f| %>
+        <% if any_errors %>
+          <div class="crayons-card crayons-card--secondary crayons-notice crayons-notice--danger my-8" role="alert" data-testid="signup-errors">
+            <div class="crayons-card__header">
+              <h3 class="crayons-card__headline">
+                Whoops, we found <%= pluralize(resource.errors.size, "problem") %>
+              </h3>
+            </div>
+            <div class="crayons-card__body">
+              <ul>
+                <% resource.errors.full_messages.each do |message| %>
+                  <li><%= message %></li>
+                <% end %>
+              </ul>
+            </div>
+          </div>
+        <% end %>
+
+        <div class="align-center">
+          <p class="pb-4 fs-3xl fw-bold lh-tight">Let's start your Forem journey!</p>
+          <p class="color-base-70 fs-xl lh-tight">Create your admin account first.</p>
+          <p class="color-base-70 fs-xl lh-tight">Then we'll walk you through your Forem setup.</p>
+        </div>
+
+        <div class="crayons-field mt-6">
+          <%= f.label :name, "Name", for: "name", class: "crayons-field__label" %>
+          <%= f.text_field :name, id: "name", placeholder: "John Doe", class: "crayons-textfield js-creator-signup-name", required: true %>
+
+          <div class="js-creator-signup-username-hint-row hidden">
+            <div class="fs-xs color-base-60 flex">
+              <span>Your username will be&nbsp;</span>
+              <button aria-label="Edit username" class="cursor-pointer border-none bg-transparent p-0 js-creator-edit-username flex color-base-90 " role="button">
+                <span class="js-creator-signup-username-hint"></span>
+                <%= inline_svg_tag("pencil.svg", class: "cursor-pointer crayons-icon opacity-75 ml-1", aria_hidden: true, title: "Edit username") %>
+              </button>
+            </div>
+            </div>
+        </div>
+
+        <div class="crayons-field mt-6 <%= "hidden" unless any_errors %> js-creator-signup-username-row">
+          <%= f.label :username, "Username", for: "username", class: "crayons-field__label" %>
+          <%= f.text_field :username, pattern: "[A-Za-z0-9_]+", maxlength: "30", title: "A username can only contain letters, numbers and underscores. The maximum username length can be 30 characters.", id: "username", class: "js-creator-signup-username crayons-textfield" %>
+          <p class="fs-xs color-base-60 mt-1">It can only contain letters, numbers and underscores. The maximum username length can be 30 characters.</p>
```

Comment:
```suggestion
          <%= f.text_field :username, pattern: "[A-Za-z0-9_]+", maxlength: "30", title: "A username can only contain letters, numbers and underscores. The maximum username length can be 30 characters.", id: "username", class: "js-creator-signup-username crayons-textfield" aria: { describedby: "username-helper-text" } %>
          <p id="username-helper-text" class="fs-xs color-base-60 mt-1">It can only contain letters, numbers and underscores. The maximum username length can be 30 characters.</p>
```

(suggestion) - linking them like this means that when a screen reader user visits the text field, their screen reader will give them the additional helper info as they're typing their username

Reply:
Ooooh this is pretty cool!